elastic/kibana
# Update Users/Hosts risk score to useSearchStrategy

## Summary

This is part of issue https://github.com/elastic/kibana/issues/129054, the original performance issue: https://github.com/elastic/kibana/pull/132703

This PR enhances the search strategy for Hosts / Users risk score, this saves a lot of re-render for the module and enhance the performance on the client side.

**Steps to verify**
- Enable the feature flag for hosts / users risk score by adding this in **kibana.dev.yml**
```xpack.securitySolution.enableExperimental: ['riskyHostsEnabled', 'riskyUsersEnabled']```
- Go to /overview page, scroll to the bottom and find the Host risk score module.
- Go to rules creation page, create a rule and generate some alerts, and back to overview page.
- Click on the `Enable via Dev Tools button`, it takes you to the dev tools
- Click on all the steps in dev tools and back to the overview page
- Observe that the data should be displayed in the module (If not please go back to the dev tools and do the restart transform steps after step 11)

- Click on hosts, host details, users, user details page, clicking around tabs and links with the console opened, no error appears
- Click a host name from an events table, and check the host flyout can be open and closed


**Hosts Risk Score Scenarios**


 **Initial state (with alerts data)**, Means the module hasn't been activated, so indices of Transform hasn't been available yet.

- Enable via Dev Tools button should be enabled.
- Import Dashboard button should be hidden.
- It should swallow the index not found error.

<img width="1675" alt="Screenshot 2022-07-28 at 16 24 46" src="https://user-images.githubusercontent.com/6295984/181576629-76fa1044-0135-4953-952c-e406f1318d6b.png">



**Module enabled but no data found** - show display the wordings as below

<img width="1664" alt="Screenshot 2022-07-29 at 17 49 42" src="https://user-images.githubusercontent.com/6295984/181807023-cf57fd3c-cdf9-4f54-ab48-77246cb74301.png">


**Module enabled** - import dashboard button should be enabled

<img width="1659" alt="Screenshot 2022-07-28 at 16 30 40" src="https://user-images.githubusercontent.com/6295984/181577807-dcb31b2d-b347-41aa-a03f-266d27d17715.png">








### Checklist

Delete any items that are not applicable to this PR.
